### DISCUSSION TITLE
Melhorias no Cuida para a próxima versão (v2.0.0)

### DISCUSSION THREAD

**Discussion by RafaelGondi** [Sysvale/cuida#144] [Nova versão]
## Discussão sobre as melhorias para a v2.0.0

Galera, vamos listar aqui sugestões de melhoria para serem lançadas na próxima versão do Cuida. Algumas já foram identificadas, mas por serem major changes e causarem incompatibilidades nas APIs dos componentes, é interessante que as agrupemos e as lancemos juntas na nova versão.

**Essa discussão ficará aberta por 15 dias**, momento a partir do qual passaremos a debater questões relativas à implementação da v2.0.0.

As sugestões acatadas serão posteriormente convertidas em issues.

Caso vocês tenham novas sugestões, objeções, observações, etc. é só colocar abaixo. o/

**Comment by RafaelGondi** [comment]
- Atualização da iconografia do Cuida para o Ionicons, removendo as famílias tipográficas antigas

**Comment by RafaelGondi** [comment]
- Adição de espaçamentos intermediários e negativos nos tokens de espaçamento

**Comment by RafaelGondi** [comment]
- Adição de shades extras para cada uma das cores definidas, aumentando a flexibilidade da paleta, sobretudo para o trabalho de ilstrações

**Comment by RafaelGondi** [comment]
- Passar os token de peso de fonte de mixin para variáveis scss para facilitar o uso

**Comment by RafaelGondi** [comment]
- Passar o conteúdo da badge por slot e não por prop.

![image](https://user-images.githubusercontent.com/20057968/111668750-d4639c80-87f4-11eb-808e-fe6132b08182.png)

**Comment by RafaelGondi** [comment]
- Passar imagem, título e botão do EmptyState como slot.

![image](https://user-images.githubusercontent.com/20057968/111668865-fbba6980-87f4-11eb-81e6-5fed4dcafeee.png)

**Comment by RafaelGondi** [comment]
- No ProgressCircular remover a prop de stroke e a substituir por uma que defina tamanhos pré-definidos, a fim de padronizar o uso

![image](https://user-images.githubusercontent.com/20057968/111669099-33291600-87f5-11eb-832b-0f40c8f3daf6.png)

**Reply by lucasn4s** [reply]
No caso dessa sugestão, já temos quais seriam os tamanhos pré-definidos ou ainda iríamos discutir?

**Reply by RafaelGondi** [reply]
Temos aquele padrão herdado do bootstrap em outros componentes do Cuida, que usa "sm", "md" e "lg", mas ainda teríamos que definir o tamanho do ProgressBar pra cada uma dessas pré-definições

**Comment by RafaelGondi** [comment]
- Analisar a unificação / generalização dos componentes [ExpansionCard](https://sysvale.github.io/cuida/?path=/docs/componentes-containers-expansioncard--expansion-card) e [Totalizer](https://sysvale.github.io/cuida/?path=/docs/componentes-containers-totalizer--totalizer) devido a sua semelhança

ExpansionCard:
![image](https://user-images.githubusercontent.com/20057968/111669394-83a07380-87f5-11eb-89c0-a520a186f22f.png)

Totalizer:
![image](https://user-images.githubusercontent.com/20057968/111669431-8c914500-87f5-11eb-88c0-a8239548fc25.png)

**Comment by RafaelGondi** [comment]
- Analisar a renomeação do componente ModalWindow

**Comment by LEMSantos** [comment]
Acredito que possa ser interessante ter um sistema de grid/flex do próprio cuida para manter a consistência principalmente nos espaçamentos entre as colunas, dado que o bootstrap-vue e nem o bootstrap oferecem essa opção.

**Reply by RafaelGondi** [reply]
O Bootstrap tem um sistema de [grid](https://getbootstrap.com/docs/4.0/layout/grid/), tem algo nele que tu acha que não está atendendo?

**Reply by LEMSantos** [reply]
O sistema de gutter não possui um espaçamento consistente, talvez seja interessante apenas mudar isso pra se adequar ao design system. Além disso só possui uma opção de espaçamento que é a que vem pré-definida alem de não ter espaçamento na parte de baixo.

**Reply by LEMSantos** [reply]
Além disso possui apenas direção em linha e não em coluna que muitas vezes é necessário para o posicionamento, fazendo com que sejam colocados elementos desnecessários como uma coluna vazia com 100% de tamanho para suprir essa necessidade. Exemplo presente na própria documentação:

```html
<div class="container">
  <div class="row">
    <div class="col">Column</div>
    <div class="col">Column</div>
    <div class="w-100"></div>
    <div class="col">Column</div>
    <div class="col">Column</div>
  </div>
</div>
```

**Comment by LEMSantos** [comment]
Em relação aos botões, seria interessante ter um mecanismo para amarrar os botões na paleta de cores padrão, também inserir a possibilidade de botões redondos, e botões com ícones.

**Comment by patriciacoelho** [comment]
Colocar um 'CopyToken' na paleta de cores

**Comment by tassiocaique** [comment]
Seria interessante avaliar o contraste texto/background das cores na paleta. Por exemplo: não sei se aquela cor de texto é a melhor opção no Turquesa-Perry base.

**Comment by patriciacoelho** [comment]
Talvez `collapsedActionName` e  `expandedActionName` sejam props desnecessárias no componente ActionsList e ficar só com slot com valor padrão e qlqr outra coisa fica dinâmico pelo slot

**Comment by patriciacoelho** [comment]
Acho que daria pra melhorar o componente StatusItem mas é uma parada tão específica que não acho que vale a pena

**Reply by RafaelGondi** [reply]
Talvez seja o caso de movermos ele do Cuida pra o CS, já que não parece ser "generalisável"

**Comment by LEMSantos** [comment]
![image](https://user-images.githubusercontent.com/23699861/111682049-ae44f900-8802-11eb-9082-792ea9bcbb1d.png)

Talvez seja interessante a adição de Chips como os exemplificados na imagem acima. Com botão de fechar também.

**Reply by RafaelGondi** [reply]
Esses caras são clicáveis?

**Reply by LEMSantos** [reply]
Podem ser, mas não é mandatório, acho que depende do contexto que seriam inseridos na aplicação, o ideal é que tenha as duas opções.

**Comment by LEMSantos** [comment]
![image](https://user-images.githubusercontent.com/23699861/111684460-8014e880-8805-11eb-9c95-f3c6c4ac42ce.png)


Uma coisa que faz falta e utilizamos bastante é um `Date Range Picker`. Seria interessante ter um que siga os padrões do design system. Alguma coisa como o da imagem acima.

**Reply by RafaelGondi** [reply]
Temos o [calendar](https://sysvale.github.io/cuida/?path=/docs/componentes-forms-calendar--calendar). Ele é um wrapper do vCalendar com algumas modificações pra atender a alguns pontos do Cuida. Tu sente a necessidade de algum recurso que não tenha lá?

**Reply by LEMSantos** [reply]
Não, acho complicado colocar ele pra fazer um filtro por exemplo teria que utilizar algo para expandir o calendário quando necessário já que ele ocupa bastante espaço na tela, mas isso pode ser só uma viagem minha também.

**Comment by johnn1sbo3s** [comment]
Seria interessante rever todos os textos para padronizar a escrita e a pontuação.

**Comment by lucasn4s** [comment]
Seria interessante refinar as nomenclaturas utilizadas nos códigos, para decidirmos o uso de português ou inglês como idioma a ser adotado.

**Comment by RafaelGondi** [comment]
Refletir sobre as releases, frequência e conteúdo. É ok continuarmos criando release pra cada PR? Ou é melhor agruparmos alguns PRs e mandar uma release com mais conteúdo? Faz sentido termos um branch developer nesse ponto?
